src/otp/otp.service.ts

```diff
@@ -3,26 +3,63 @@ import { JsonContains, MoreThan, Repository } from "typeorm";
 import { Injectable } from "@nestjs/common";
 import { Otp } from "../entities";
 import { InjectRepository } from "@nestjs/typeorm";
+import { SendOtpResult } from "./dto";
+import { uid } from "uid";
+import { ConfigService } from "@nestjs/config";
 
 @Injectable()
 export class OtpService {
   constructor(
     @InjectRepository(Otp)
-    private readonly otpRepository: Repository<Otp>
+    private readonly otpRepository: Repository<Otp>,
+    private readonly configService: ConfigService
   ) {}
 
   async validateEmailOtpAsync(
     email: string,
-    otpCode: string
+    otpValue: string
   ): Promise<boolean> {
     const otpEntity = await this.otpRepository.findOne({
       where: {
-        value: otpCode,
+        value: otpValue,
         channel: JsonContains({ email }),
         expiresAt: MoreThan(new Date()),
       },
     });
 
     return !!otpEntity;
   }
+  private async createOtpAsync(channel: {
```

Comment:
Move private methods to bottom to see public methods first when maintaining the service.